packages/-ember-data/tests/acceptance/tracking-model-id-test.js

```diff
@@ -0,0 +1,101 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, settled } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import Store from '@ember-data/store';
+import Model, { attr } from '@ember-data/model';
+import JSONAPIAdapter from '@ember-data/adapter/json-api';
+import JSONAPISerializer from '@ember-data/serializer/json-api';
+import Component from '@glimmer/component';
+import { dependentKeyCompat } from '@ember/object/compat';
+import { gte } from 'ember-compatibility-helpers';
+import { has } from 'require';
+import { resolve } from 'rsvp';
+
+if (gte('3.14.0') && has('@glimmer/component')) {
+  class Widget extends Model {
+    @attr() name;
+
+    @dependentKeyCompat
```

Comment:
This shouldn't be necessary it looks like, nothing is depending on this property directly. This would be necessary if you had:

```js
@computed('numericId')
```

Somewhere, or an observer that was trying to watch it. The template layer does not use observers or computeds anymore.